core/trino-main/src/main/java/io/trino/metadata/TableProperties.java

```diff
@@ -82,11 +77,13 @@ public Optional<DiscretePredicates> getDiscretePredicates()
     {
         private final PartitioningHandle partitioningHandle;
         private final List<ColumnHandle> partitioningColumns;
+        private final boolean singleSplitPerPartition;
```

Comment:
add a comment that it only implies that single partition will not be present in more than one split. Single split can still hold multiple partitions. It is not super obious (at least to me) just based on the variable name.

Reply:
Yeah, I also realized that it doesn't represent the true nature of it. Ideally it would be great to name it right. I was trying to find a better name, but found it difficult. Open for suggestions :-)

Reply:
Can you describe what this flag is meant to represent?

Reply:
I tried to find a better name, but couldn't come up with anything better. This flag indicates that that a certain unique tuple of `partitioningColumns` values will be present in exactly one split. This way the engine allowed to flush intermediate state of an aggregation at split boundaries.

Reply:
What's odd about this is that partitioning is (or was, at some point in the past) already supposed to imply that. That's what allows the engine to use that partitioning columns information to skip operations that would otherwise have to enforce that partitioning.

Reply:
Currently when `private final List<ColumnHandle> partitioningColumns;` is not empty it indicates that the data is partitioned by a certain set of columns. However by default (when this flag is set to `false`) a single partition may appear spread out across multiple splits of the same group (bucket) of splits. The group (bucket) a certain split belongs to is defined by https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorNodePartitioningProvider.java#L29. 

This flag set to `true` implies a stronger guarantee. It effectively indicates that the number of splits for a certain bucket will not be larger than 1, what allows the engine to avoid unnecessary local shuffles as it is guaranteed that only a single split will contain all the values for a certain "partition".